android/src/main/kotlin/io/flutter/plugins/flutterexifrotation/FlutterExifRotationPlugin.kt

```diff
@@ -10,17 +10,17 @@ import io.flutter.embedding.engine.plugins.FlutterPlugin
 import io.flutter.plugin.common.MethodCall
 import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import kotlinx.coroutines.*
 import java.io.File
 import java.io.FileOutputStream
 import java.io.IOException
-import java.util.concurrent.ExecutorService
-import java.util.concurrent.Executors
 
 /**
  * FlutterExifRotationPlugin
  */
 class FlutterExifRotationPlugin : FlutterPlugin, MethodCallHandler {
     private var applicationContext: Context? = null
+    private val mainScope = CoroutineScope(Dispatchers.Main)
```

Comment:
We should avoid using `Dispatchers.Main` for long operations. 
IO or CPU is better for this. Because we are reading data from a disk and then we are manipulating this data and writing it.
But, we want to fix the `OutOfMemoryException`, so we need to prevent new Threads creation. So single thread scheduler should be most suitable for the current case.

But, as we decide to use only one thread, do we need coroutines here?

Reply:
@antondudakov single thread is a bit slow in my  manual testing but we received couple ANRs with Dispatchers.Main so changing to your suggestion 👍